### PR TITLE
Addition of trim segment configuration and new documentation

### PR DESCRIPTION
**Pull request by NgoniChoga**
This adds [the feature defined in this issue](https://github.com/opencast/opencast/issues/1351) of no trim segments at the start and at the end of the video

### CODE REVIEW

**Comment by lkiesow** [comment]
The build failure is caused by the not properly integrated documentation page.
See Travis CI build logs for more details.

**Comment by gregorydlogan** [comment]
I think the changes to the admin UI have cause more problems again @NgoniChoga, apologies!

File: modules/admin-ui-frontend/app/scripts/shared/directives/timelineDirective.js [hunk 1]
```diff
@@ -1264,6 +1264,122 @@ angular.module('adminNg.directives')
...
+              return (_s.end - _s.start);
+            };
+
+            var maxSegIdx = segments.length - 1;
+            segments[0].start = 0;
+            segments[maxSegIdx].end = duration;
+
+            if (segments.length == 1) {
+
+              var _start = 0, _duration = duration;
```
Comment (gregorydlogan) [inline modules/admin-ui-frontend/app/scripts/shared/directives/timelineDirective.js]: We don't really enforce code standards in the javascript, but please declare each variable on a separate line.

File: modules/admin-ui-frontend/app/scripts/shared/directives/timelineDirective.js [hunk 2]
```diff
@@ -1264,6 +1264,122 @@ angular.module('adminNg.directives')
...
+              };
+            };
+
+            var getSegmentLength = function(_s) {
+              return (_s.end - _s.start);
+            };
+
+            var maxSegIdx = segments.length - 1;
+            segments[0].start = 0;
+            segments[maxSegIdx].end = duration;
```
Comment (gregorydlogan) [inline modules/admin-ui-frontend/app/scripts/shared/directives/timelineDirective.js:1312]: If there's only one segment, doesn't `maxSegIdx` end up at -1?

File: modules/admin-ui-frontend/app/scripts/shared/directives/timelineDirective.js [hunk 3]
```diff
@@ -1264,6 +1264,122 @@ angular.module('adminNg.directives')
           if (scope.timer) $timeout.cancel( scope.timer );
         });
 
+        /**
+         * Creates a promise object to load the AuthService so that user properties
+         * are available to the current scope.
+         */
+        AuthService.getUser().$promise.then(function (user) {
```
Comment (gregorydlogan) [inline modules/admin-ui-frontend/app/scripts/shared/directives/timelineDirective.js:1271]: I'm not a huge fan of the way this has nested function declarations.  Is this how the rest of Angular codebase is?  It's been a while, but we try to keep things similar, file to file.

**Review by gregorydlogan — COMMENTED** [review]
Appears to work as expected.  Minor concerns as described.

**Comment by gregorydlogan** [comment]
@NgoniChoga ping

File: modules/admin-ui-frontend/app/scripts/shared/directives/timelineDirective.js (hunk 1 above)
Comment (NgoniChoga) [reply modules/admin-ui-frontend/app/scripts/shared/directives/timelineDirective.js]: Updated

File: modules/admin-ui-frontend/app/scripts/shared/directives/timelineDirective.js (hunk 2 above)
Comment (NgoniChoga) [reply modules/admin-ui-frontend/app/scripts/shared/directives/timelineDirective.js:1312]: maxSegIdx end up at 0 if there's only one segment.

File: modules/admin-ui-frontend/app/scripts/shared/directives/timelineDirective.js (hunk 3 above)
Comment (NgoniChoga) [reply modules/admin-ui-frontend/app/scripts/shared/directives/timelineDirective.js:1271]: This is how it was done in a number of different files.

**Pull request merged**